File: kiosk-app/lib/models/payment.dart
class Payment {
  final int id;
  final int orderId;
  final String method;
  final int amount;
  final String status;
  final String? approvalNumber;
  final String createdAt;

  Payment({
    required this.id,
    required this.orderId,
    required this.method,
    required this.amount,
    required this.status,
    this.approvalNumber,
    required this.createdAt,
  });

  factory Payment.fromJson(Map<String, dynamic> json) => Payment(
        id: json['id'] as int,
        orderId: json['orderId'] as int,
        method: json['method'] as String,
        amount: json['amount'] as int,
        status: json['status'] as String,
        approvalNumber: json['approvalNumber'] as String?,
        createdAt: json['createdAt'] as String,
      );
}

File: kiosk-backend/src/main/kotlin/com/kiosk/domain/model/Payment.kt
package com.kiosk.domain.model

import java.time.LocalDateTime

enum class PaymentMethod {
    CARD,
    KAKAO_PAY,
    NAVER_PAY,
    CASH
}

enum class PaymentStatus {
    PENDING,
    COMPLETED,
    CANCELLED,
    FAILED
}

data class Payment(
    val id: Long = 0,
    val orderId: Long,
    val method: PaymentMethod,
    val amount: Int,
    val status: PaymentStatus = PaymentStatus.PENDING,
    val approvalNumber: String? = null,
    val pgTransactionId: String? = null,
    val createdAt: LocalDateTime = LocalDateTime.now()
)
